File: app/src/main/java/com/prongbang/localizedapp/MainActivity.java
package com.prongbang.localizedapp;

import android.content.Intent;
import android.content.res.Configuration;
import android.os.Bundle;
import android.support.annotation.LayoutRes;
import android.util.Log;
import android.view.View;
import android.widget.ImageView;

import com.prongbang.localization.base.LocalizationAppCompatActivity;
import com.prongbang.localization.utils.LocaleHelper;
import com.prongbang.localizedapp.views.SettingActivity;

public class MainActivity extends LocalizationAppCompatActivity {

    private String TAG = MainActivity.class.getSimpleName();

    private ImageView ivSetting;

    private boolean USE_EXAMPLE_2 = false;

    @Override
    protected void onCreate(final Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // Ex2
        if (USE_EXAMPLE_2) {
            // bind view
            bindSetting();
        }
    }

    @Override
    public void setContentView(@LayoutRes final int layoutResID) {
        super.setContentView(layoutResID);

        if (!USE_EXAMPLE_2) {

            // bind view
            bindSetting();

            Log.i(TAG, "setContentView: Reload");
        }
    }

    private void bindSetting() {
        ivSetting = findViewById(R.id.ivSetting);
        ivSetting.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this, SettingActivity.class);
                startActivity(intent);
            }
        });
    }

    @Override
    public void onConfigurationChanged(Configuration newConfig) {
        super.onConfigurationChanged(newConfig);
        Log.i(TAG, "onConfigurationChanged: ");

        // Ex1
        if (!USE_EXAMPLE_2) {
            setContentView(R.layout.activity_main);
        } else {
            // Ex2
            recreate();
        }
    }
}

File: app/src/main/java/com/prongbang/localizedapp/MainActivity.kt
package com.prongbang.localizedapp

import android.os.Bundle
import com.prongbang.localization.ENGLISH
import com.prongbang.localization.LocalizationAppCompatActivity
import com.prongbang.localization.Localize
import com.prongbang.localization.LocalizeManager
import com.prongbang.localization.THAI
import com.prongbang.localizedapp.databinding.ActivityMainBinding
import com.prongbang.localizedapp.feed.Feed
import com.prongbang.localizedapp.feed.FeedActivity
import com.prongbang.localizedapp.feed.FeedAdapter

class MainActivity : LocalizationAppCompatActivity() {

	private val binding: ActivityMainBinding by lazy { ActivityMainBinding.inflate(layoutInflater) }
	private val feedAdapter by lazy { FeedAdapter() }

	override fun onCreate(savedInstanceState: Bundle?) {
		super.onCreate(savedInstanceState)
		setContentView(binding.root)
		initView()
		initListener()
	}

	override fun onResume() {
		super.onResume()
		loadFeed()
	}

	private fun initView() {
		binding.feedRecyclerView.apply {
			adapter = feedAdapter
		}
		feedAdapter.apply {
			onItemClick = {
				FeedActivity.navigate(this@MainActivity, it)
			}
		}
	}

	private fun initListener() {
		binding.ivSetting.setOnClickListener {
			SettingActivity.navigate(this)
		}
	}

	private fun loadFeed() {
		val locale = LocalizeManager.getLocale(this)
		when (locale.language) {
			Localize.THAI.language -> {
				feedAdapter.submitList(Feed.asListTH())
			}
			Localize.ENGLISH.language -> {
				feedAdapter.submitList(Feed.asListEN())
			}
		}
	}

}